airbyte-commons-worker/src/main/java/io/airbyte/workers/internal/AirbyteMessageTracker.java

```diff
@@ -266,8 +277,35 @@ private void handleEmittedErrorTrace(final AirbyteTraceMessage errorTraceMessage
     }
   }
 
-  @SuppressWarnings("PMD") // until method is implemented
-  private void handleEmittedEstimateTrace(final AirbyteTraceMessage estimateTraceMessage, final ConnectorType connectorType) {
+  /**
+   * There are several assumptions here:
+   * <p>
+   * - Assume the estimate is a whole number and not a sum i.e. each estimate replaces the previous
```

Comment:
Isn't this putting the burden of tracking the estimate on the connector?
example: I started the sync, initial estimate is 10k rows, during the sync, there's 5k now rows that were added. I understand that the source connector should emit (10k + 15k - the number of rows that has been sent)

Reply:
yeap - this is something optional connectors can choose to implement. Were you thinking of better ways to do this?

Reply:
I'd opt for a way that doesn't involve tracking how many rows were sent by the connector.

I think the easiest contract would be "how many records needs to be replicated since the start where start would be the state being sent to the source".
This has the benefit of decoupling some information:
* we get from the source how many records needs to be sent.
* the platform can track how many records were received from the source
* the platform can track how many records were persisted by the destination
* we can choose if we want to report progress purely from the source and eventually report progress with what has actually been committed by the destination.

Reply:
There is misunderstanding here. That is the contract - the connector sends how many records it expects to need to replicate when the sync starts. 

If this estimate changes, the connector can send a new message to the platform. The new number overrides the previous estimate.

Probably easier to talk really quickly than type this out.

Reply:
Awesome. The part where I got confused was if we expected the connector to track the emitted messages. As long as that's not part of the expectations, whole number or sum feels like implementation details, whole number is probably easier since it doesn't involve keeping track of what was sent.